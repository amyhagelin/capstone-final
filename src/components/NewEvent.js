import React from "react";
import {connect} from 'react-redux';

import { newEvent } from "../actions/newEvent";

export class NewEvent extends React.Component {

    handleNewEvent = (event) => {
        event.preventDefault();
        console.log(this.props);
        const dateValue = this.dateInput.value;
        const timeValue = this.timeInput.value;
		const typeValue = this.typeInput.value;
		const triggersValue = this.triggersInput.value;
		const locationValue = this.locationInput.value;
		const medicationValue = this.medicationInput.value;
		const notesValue = this.notesInput.value;
        this.props.dispatch(newEvent(
			dateValue, 
			timeValue,
			typeValue,
			locationValue,
			triggersValue,
			medicationValue,
			notesValue
			));
    }

    render() {
        return (
            <div>
				<h2>Add New Event</h2>
              <form onSubmit={ this.handleNewEvent }>
				<div className="inline-block">
				<input placeholder="DATE" type="date" ref={ input => this.dateInput = input } /><br />
				<input placeholder="TIME" type="text" ref={ input => this.timeInput = input } /><br />
				<select name="type" ref={ input => this.typeInput = input }>
					<option value="" disabled selected>SEIZURE TYPE</option>
					<option value="aura">Aura</option>
		 			<option value="absence">Absence</option>
		  			<option value="atonic-">Atonic</option>
		 			<option value="clonic">Clonic</option>
		 			<option value="myoclonic">Myoclonic</option>
		  			<option value="tonic">Tonic</option>
		  			<option value="tonic-clonic">Tonic-Clonic</option>
	  			</select>
	  			<br />
				<input placeholder="TRIGGERS" type="text" ref={ input => this.triggersInput = input } /><br />
				</div>
				<div className="inline-block">
				<input placeholder="LOCATION" type="text" ref={ input => this.locationInput = input } /><br />
				<input placeholder="CURRENT MEDICATION" type="text" ref={ input => this.medicationInput = input } /><br />
				<input placeholder="NOTES"type="text" ref={ input => this.notesInput = input } /><br />
				<button className="button">Save</button>
				</div>
			    </form>
				<div className="alert">
					{ this.props.alert && this.props.alert.title }
				</div>
            </div>
        );
    }

};

const mapStateToProps = state => ({
    alert: state.ui.alert
});

export default connect(mapStateToProps)(NewEvent);
