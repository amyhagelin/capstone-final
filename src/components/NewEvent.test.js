import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import { NewEvent } from './NewEvent';

describe ('<NewEvent />', () => {

    it('renders correctly', () => {
         const event = {
            date: new Date(123),
            time: '6:00 pm',
            type: 'aura',
            triggers: 'xyz',
            location: 'home',
            medication: 'abc',
            notes: 'def'
        };
        const tree = renderer.create(
            <NewEvent event={ event } />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should dispatch action when clicking submit', () => {
        const dispatch = jest.fn()
        const wrapper = mount(<NewEvent dispatch={ dispatch }/>);
        wrapper.find('form').simulate('submit');
        expect(dispatch).toHaveBeenCalled();
    });

});