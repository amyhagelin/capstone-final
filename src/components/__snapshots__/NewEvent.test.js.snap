// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewEvent /> renders correctly 1`] = `
<div>
  <h2>
    Add New Event
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div
      className="inline-block"
    >
      <input
        placeholder="DATE"
        type="date"
      />
      <br />
      <input
        placeholder="TIME"
        type="text"
      />
      <br />
      <select
        name="type"
      >
        <option
          disabled={true}
          selected={true}
          value=""
        >
          SEIZURE TYPE
        </option>
        <option
          value="Aura"
        >
          Aura
        </option>
        <option
          value="Absence"
        >
          Absence
        </option>
        <option
          value="Atonic-"
        >
          Atonic
        </option>
        <option
          value="Clonic"
        >
          Clonic
        </option>
        <option
          value="Myoclonic"
        >
          Myoclonic
        </option>
        <option
          value="Tonic"
        >
          Tonic
        </option>
        <option
          value="Tonic-clonic"
        >
          Tonic-Clonic
        </option>
      </select>
      <br />
      <input
        placeholder="TRIGGERS"
        type="text"
      />
      <br />
    </div>
    <div
      className="inline-block"
    >
      <input
        placeholder="LOCATION"
        type="text"
      />
      <br />
      <input
        placeholder="CURRENT MEDICATION"
        type="text"
      />
      <br />
      <input
        placeholder="NOTES"
        type="text"
      />
      <br />
      <button
        className="button"
      >
        Save
      </button>
    </div>
  </form>
  <div
    className="alert"
  />
</div>
`;
